File: python/testopmodes/spin_shooter.py
# Copyright (c) FIRST and other WPILib contributors.
# Open Source Software; you can modify and/or share it under the terms of
# the WPILib BSD license file in the root directory of this project.

from commands3 import Command, Scheduler
from wpilib.opmode import PeriodicOpMode, test_op_mode


@test_op_mode
class SpinShooter(PeriodicOpMode):
    """Test mode that continuously spins the shooter at 40 RPM."""

    def __init__(self, robot):
        super().__init__()
        self._robot = robot
        self._spin_command = robot.getShooter().getSpinCommand()

    def disabledPeriodic(self):
        self._robot.robotPeriodic()

    def start(self):
        Scheduler.getDefault().schedule(self._spin_command)

    def periodic(self):
        self._robot.robotPeriodic()

    def end(self):
        Scheduler.getDefault().cancel(self._spin_command)

File: java/src/main/java/frc/robot/testopmodes/SpinShooter.java
package frc.robot.testopmodes;

import org.wpilib.command3.Command;
import org.wpilib.command3.Scheduler;
import org.wpilib.opmode.PeriodicOpMode;
import org.wpilib.opmode.TestOpMode;

import frc.robot.Robot;

@TestOpMode
public class SpinShooter extends PeriodicOpMode {
    private final Robot m_robot;

    private final Command spinCommand;

    public SpinShooter(Robot robot) {
        m_robot = robot;
        spinCommand = m_robot.getShooter().getSpinCommand();
    }

    @Override
    public void disabledPeriodic() {
        m_robot.robotPeriodic();
    }

    @Override
    public void start() {
        Scheduler.getDefault().schedule(spinCommand);
    }

    @Override
    public void periodic() {
        m_robot.robotPeriodic();
    }

    @Override
    public void end() {
        Scheduler.getDefault().cancel(spinCommand);
    }

}
